### ISSUE TITLE
 Error inflating class com.dimelo.dimelosdk.utilities.DetectKeyboardCoordinatorLayout

### ISSUE THREAD

**Issue by maresc-g** [dimelo/Dimelo-Android#1]
Hi,

I have an error when I try to use the version 1.1 of the SDK. The app crashes on launch everytime with the error  `Error inflating class com.dimelo.dimelosdk.utilities.DetectKeyboardCoordinatorLayout`.
Here is the stacktrace :

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{com.pictime.electrodepot/com.pictime.electrodepot.activity.ActivityContact}: android.view.InflateException: Binary XML file line #2: Binary XML file line #2: Error inflating class com.dimelo.dimelosdk.utilities.DetectKeyboardCoordinatorLayout
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2426)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2490)
    at android.app.ActivityThread.-wrap11(ActivityThread.java)
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1354)
    at android.os.Handler.dispatchMessage(Handler.java:102)
    at android.os.Looper.loop(Looper.java:148)
    at android.app.ActivityThread.main(ActivityThread.java:5443)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618)
Caused by: android.view.InflateException: Binary XML file line #2: Binary XML file line #2: Error inflating class com.dimelo.dimelosdk.utilities.DetectKeyboardCoordinatorLayout
    at android.view.LayoutInflater.inflate(LayoutInflater.java:539)
    at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
    at com.dimelo.dimelosdk.main.Chat.onCreateView(Chat.java:524)
    at android.support.v4.app.Fragment.performCreateView(Fragment.java:1974)
    at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1067)
    at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1252)
    at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:738)
    at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1617)
    at android.support.v4.app.FragmentController.execPendingActions(FragmentController.java:339)
    at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:602)
    at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1260)
    at android.app.Activity.performStart(Activity.java:6261)
    at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2389)
    at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2490) 
    at android.app.ActivityThread.-wrap11(ActivityThread.java) 
    at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1354) 
    at android.os.Handler.dispatchMessage(Handler.java:102) 
    at android.os.Looper.loop(Looper.java:148) 
    at android.app.ActivityThread.main(ActivityThread.java:5443) 
    at java.lang.reflect.Method.invoke(Native Method) 
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:728) 
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:618) 
```

I did add 

```
        renderscriptTargetApi 23
        renderscriptSupportModeEnabled true
```

to my gradle file.

I managed to make it work with the version 1.0 but is there a way to make it work with the 1.1 ?

Thanks

**Comment by elhu** [comment]
Hi,

Have you tried forcing a full-refresh of your Gradle dependencies by running 

```
./gradlew clean && ./gradlew --refresh-dependencies
```

from within your app directory? It might sort out this kind of issues.

If this does not help, I'm going to need more info, starting with the Android version running on the device.

Let me know if this helped!

**Comment by maresc-g** [comment]
It didn't work. I tried on several devices, a Motorola moto g on 6.0, a LG Nexus 5 on 4.4.4 and a Samsung Galaxy S3 on 4.3

Here is my build.gradle :

```
apply plugin: 'com.android.library'
apply plugin: 'io.fabric'

buildscript {
    repositories {
        mavenCentral()
        maven { url 'https://maven.fabric.io/public' }
        maven { url 'https://raw.github.com/dimelo/Dimelo-Android/master' }
    }
    dependencies {
        classpath 'io.fabric.tools:gradle:1.+'
    }
}

repositories {
    mavenCentral()
    maven { url 'https://maven.fabric.io/public' }
    maven { url 'https://raw.github.com/dimelo/Dimelo-Android/master' }
}

android {
    compileSdkVersion 23
    buildToolsVersion '23.0.2'
    useLibrary 'org.apache.http.legacy'
    defaultConfig {
        minSdkVersion 15
        targetSdkVersion 21
        multiDexEnabled true
        renderscriptTargetApi 23
        renderscriptSupportModeEnabled true
    }

    buildTypes {
        release {
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
        debug{
            minifyEnabled false
            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
        }
    }

    lintOptions {
        disable 'InvalidPackage'
    }
}

dependencies {
    compile fileTree(dir: 'libs', include: '*.jar')
    compile 'com.dimelo.dimelosdk:dimelosdk:1.1.1'
    compile 'com.android.support:appcompat-v7:23.2.1'
    compile 'com.google.android.gms:play-services-maps:8.1.0'
    compile('com.crashlytics.sdk.android:crashlytics:2.5.5@aar') {
        transitive = true;
    }
    compile 'com.google.code.gson:gson:2.2.4'
}
```

**Comment by elhu** [comment]
Thank you for those details.

I see that you are using `proguard`. Could you try adding the following to your proguard configuration file, and let me know if it solves your problem?

```
-keep class com.dimelo.** { *; }
-keep interface com.dimelo.** { *; }
```

**Comment by maresc-g** [comment]
I'm not using proguard this is the default configuration, proguard is disabled by `minifyEnabled false`

**Comment by elhu** [comment]
Indeed, I spoke too quickly.

The problem might come from a configuration issue with multidex.
Is Multidex configured in your Manifest file, as explained in http://developer.android.com/tools/building/multidex.html#mdex-gradle ?

If you've already subclassed `Application`, you might want to have a look at http://stackoverflow.com/questions/26609734/how-to-enable-multidexing-with-the-new-android-multidex-support-library.

**Comment by maresc-g** [comment]
It wasn't configured in the AndroidManifest but adding it doesn't change anything.
I also tried to upgrade the dimelo sample app (https://github.com/dimelo/Dimelo-Android-SampleApp) to use the 1.1 SDK version but I get a compile error :

```
Error:duplicate files during packaging of APK C:\datas\marescauxg\AndroidStudioProjects\Dimelo-Android-SampleApp-master\Dimelo-Android-SampleApp-master\app\build\outputs\apk\app-debug-unaligned.apk
    Path in archive: lib/armeabi-v7a/librsjni.so
    Origin 1: C:\datas\marescauxg\android-sdk-windows\build-tools\23.0.1\renderscript\lib\packaged\armeabi-v7a\librsjni.so
    Origin 2: C:\datas\marescauxg\AndroidStudioProjects\Dimelo-Android-SampleApp-master\Dimelo-Android-SampleApp-master\app\build\intermediates\exploded-aar\com.dimelo.dimelosdk\dimelosdk\1.1.0\jni\armeabi-v7a\librsjni.so
You can ignore those files in your build.gradle:
    android {
      packagingOptions {
        exclude 'lib/armeabi-v7a/librsjni.so'
      }
    }
Error:Execution failed for task ':app:packageDebug'.
> Duplicate files copied in APK lib/armeabi-v7a/librsjni.so
    File 1: C:\datas\marescauxg\android-sdk-windows\build-tools\23.0.1\renderscript\lib\packaged\armeabi-v7a\librsjni.so
    File 2: C:\datas\marescauxg\AndroidStudioProjects\Dimelo-Android-SampleApp-master\Dimelo-Android-SampleApp-master\app\build\intermediates\exploded-aar\com.dimelo.dimelosdk\dimelosdk\1.1.0\jni\armeabi-v7a\librsjni.so
```

**Comment by elhu** [comment]
I have updated the sample app to use SDK 1.1.1.
Could you try pulling the new code, and see if you manage to build it?

**Comment by maresc-g** [comment]
It works ! I managed to build and launch the sample app so I compared every file you modified with your commit.
I saw that in TabStart you use openChatActivity() instead of a fragment so I tried that and I applied a them to ChatActivity.
I got an error because I didn't use a compatible theme so I changed it and it worked.
So I just applied the theme to the whole application and it worked.

Thank you for your help !
